### PR TITLE
Add correlation logs for timeouts 

### PR DESCRIPTION
**Pull request by sachithKay**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: modules/core/src/main/java/org/apache/synapse/core/axis2/Axis2FlexibleMEPClient.java [hunk 1]
```diff
@@ -142,6 +144,14 @@ public static void send(
             headers.put("Cookie", session);
         }
 
+        if (originalInMsgCtx.isPropertyTrue(PassThroughConstants.CORRELATION_LOG_STATE_PROPERTY)) {
+           if (originalInMsgCtx.getProperty(PassThroughConstants.CORRELATION_ID) == null){
```
Comment (malakaganga) [inline modules/core/src/main/java/org/apache/synapse/core/axis2/Axis2FlexibleMEPClient.java]: Shall we fix the formatting issue here?

File: modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/TargetRequest.java [hunk 2]
```diff
@@ -394,6 +393,8 @@ public int write(NHttpClientConnection conn, ContentEncoder encoder) throws IOEx
     public boolean hasEntityBody() {
         return hasEntityBody;
     }
+
+    public URL getUrl(){return url; }
```
Comment (malakaganga) [inline modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/TargetRequest.java]: Shall we remove the extra space here?

File: modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/config/PassThroughConfiguration.java [hunk 3]
```diff
@@ -143,6 +143,11 @@ public int getMaximumConnectionLifespan() {
         return getIntProperty(PassThroughConfigPNames.MAXIMUM_CONNECTION_LIFESPAN, Integer.MAX_VALUE);
     }
 
+    public String getCorrelationHeaderName(){
```
Comment (malakaganga) [inline modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/config/PassThroughConfiguration.java]: Shall we fix the formatting issue here?

File: modules/core/src/main/java/org/apache/synapse/core/axis2/Axis2FlexibleMEPClient.java (hunk 1 above)
Comment (sachithKay) [reply modules/core/src/main/java/org/apache/synapse/core/axis2/Axis2FlexibleMEPClient.java]: fixed in https://github.com/wso2/wso2-synapse/pull/1305

File: modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/TargetRequest.java (hunk 2 above)
Comment (sachithKay) [reply modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/TargetRequest.java]: fixed in https://github.com/wso2/wso2-synapse/pull/1305

File: modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/config/PassThroughConfiguration.java (hunk 3 above)
Comment (sachithKay) [reply modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/config/PassThroughConfiguration.java]: fixed in https://github.com/wso2/wso2-synapse/pull/1305

**Pull request merged**